x/observer/keeper/ballot.go

```diff
@@ -82,28 +82,58 @@ func (k Keeper) AddBallotToList(ctx sdk.Context, ballot types.Ballot) {
 	k.SetBallotList(ctx, &list)
 }
 
-// ClearMaturedBallotsAndBallotList deletes all matured ballots and the list of ballots for a given height.
-// It also emits an event for each ballot deleted.
-func (k Keeper) ClearMaturedBallotsAndBallotList(ctx sdk.Context, maturityBlocksParam int64) {
-	maturedBallotsHeight := getMaturedBallotHeight(ctx, maturityBlocksParam)
-
-	// Fetch all the matured ballots, return if no matured ballots are found
-	// For the current implementation, this should never happen as ClearMaturedBallotsAndBallotList is only called after the Distribution of the rewards,
-	//	which means that there are matured ballots to be deleted
+// ClearFinalizedMaturedBallots deletes all matured and finalized ballots for a given height.
```

Comment:
Instead of two functions, it seems to me that we could have just one taking a bool in parameter, `deleteNonFinalize`